### PR TITLE
Extend `PrepareBatchExtraInput` to work with HoVerNet

### LINKED ISSUE #5028: Make HoVerNet  work SuperviseTrainer
Using `SupervisedTrainer` for HoVerNet pipeline training requires a `prepare_batch` to prepare data differently than how it is being prepared in `default_prepare_batch` since HoVerNet target includes three different tensors.

### PR DESCRIPTION
**Pull request by KumoLiu**
Signed-off-by: KumoLiu <yunl@nvidia.com>

Fixes #5028.

### Description
Since the output for HoVerNet is a dictionary, we already have `PrepareBatchExtraInput` to support extra input data for the network, but it still can't meet the requirements. This PR is to extend `PrepareBatchExtraInput` to make the label can be a dictionary.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/pathology/engines/utils.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import torch
+
+from monai.engines import PrepareBatch, PrepareBatchExtraInput
+from monai.utils import ensure_tuple
+from monai.utils.enums import HoVerNetBranch
+
+__all__ = ["PrepareBatchWithDictLabel"]
+
+
+class PrepareBatchWithDictLabel(PrepareBatch):
```
Comment (drbeh) [inline monai/apps/pathology/engines/utils.py:23]: It seems to be very specific to hovernet, so might be better to change the name to `PrepareBatchHoVerNet`. Thanks

Comment (Nic-Ma) [reply monai/apps/pathology/engines/utils.py:23]: I agree, actually I also raised this concern to @KumoLiu offline.

Thanks.

Comment (KumoLiu) [reply monai/apps/pathology/engines/utils.py:23]: Hi @drbeh @Nic-Ma, thanks for your suggestion, I will update it in the latest commit.

File: tests/test_prepare_batch_hovernet [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) MONAI Consortium
```
Comment (Nic-Ma) [inline tests/test_prepare_batch_hovernet:1]: Why this file is not a python (.py) file?

Thanks.

Comment (KumoLiu) [reply tests/test_prepare_batch_hovernet]: Thanks @Nic-Ma, I just made a mistake here, have fixed it.

**Review by Nic-Ma — APPROVED** [review]
